services/QuillLMS/spec/controllers/teacher_infos_controller_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: teacher_infos
+#
+#  id                  :bigint           not null, primary key
+#  minimum_grade_level :integer
+#  maximum_grade_level :integer
+#  teacher_id          :bigint
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+require 'rails_helper'
+
+describe TeacherInfosController do
+  let(:user) { create(:user) }
+  let(:subject_area1) { create(:subject_area) }
+  let(:subject_area2) { create(:subject_area)}
+
+  before do
+    allow(controller).to receive(:current_user) { user }
+  end
+
+  describe '#create' do
+    it 'should create a teacher info record with the data populated' do
+      post :create, params: {minimum_grade_level: 4, maximum_grade_level: 12, subject_areas: [subject_area1.id]}
+
+      teacher_info = TeacherInfo.find_by(teacher: user)
+
+      expect(teacher_info.minimum_grade_level).to eq(4)
+      expect(teacher_info.maximum_grade_level).to eq(12)
+      expect(teacher_info.subject_areas).to eq([subject_area1])
+      expect(response.status).to be(200)
+    end
+  end
+
+  describe '#update' do
```

Comment:
NB.  I opted for `before { teacher_info.subject_areas.push(subject_area1) }`
instead of  `let!(:teacher_info_subject_area) { create(:teacher_info_subject_area, teacher_info: teacher_info, subject_area: subject_area1) }` because `teacher_info_subject_area` isn't used in the test anywhere.